### PR TITLE
revert TSTORE opcode if in static context

### PR DESCRIPTION
**Pull request by moodysalem**
i just pushed this update to the EIP specification

https://eips.ethereum.org/EIPS/eip-1153#specification

### CODE REVIEW

File: packages/vm/src/evm/opcodes/functions.ts [hunk 1]
```diff
@@ -888,6 +888,9 @@ export const handlers: Map<number, OpHandler> = new Map([
   [
     0xb4,
     function (runState) {
+      if (runState.eei.isStatic()) {
```
[the remark this answers is not in the document]
Comment (moodysalem) [reply packages/vm/src/evm/opcodes/functions.ts:891]: realizing the implementation should also be updated to use the warm SSTORE of a dirty slot gas schedule instead of a fixed 100 gas (and same with SLOAD)

**Comment by moodysalem** [comment]
realized there's a bug--on commit of transient storage (ie. a call returns), we should merge the changes of the inner call context with the latest changeset. on revert, all the changes on inner calls should also revert.

we need a good number of additional evm tests to cover this case as well.

**Comment by jochem-brouwer** [comment]
In that case, we should do a thorough review of this PR (after the bugs are fixed) to ensure we have covered the edge cases as well.

**Comment by moodysalem** [comment]
> In that case, we should do a thorough review of this PR (after the bugs are fixed) to ensure we have covered the edge cases as well.

Agree, and we may want to change the implementation or limit the number of keys in the EIP per https://github.com/ethereum/pm/issues/500#issuecomment-1085966888

Sorry for the churn, this is the first implementation so working out some kinks

**Comment by jochem-brouwer** [comment]
No problem. Instead of limiting max tstore keys, what about doing the same thing as memory (quadratic costs for size). I think a limit could have nasty practical consequences.

**Comment by holgerd77** [comment]
@moodysalem let me know how this work would affect a planned release (wait? release nevertheless?).

**Comment by moodysalem** [comment]
> @moodysalem let me know how this work would affect a planned release (wait? release nevertheless?).

I wouldn't necessarily block release, the only risk is that someone prototyping runs into it

**Comment by holgerd77** [comment]
Opened up for merge after chat discussion with @tynes, so will set this as "Ready for Review" and merge in ones tests pass.

**Review by holgerd77 — APPROVED** [review]
LGTM

**Comment by moodysalem** [comment]
> No problem. Instead of limiting max tstore keys, what about doing the same thing as memory (quadratic costs for size). I think a limit could have nasty practical consequences.

Agree with you here... need to think more about what makes sense. Regular storage doesn't have this issue because each new slot costs 2 orders of magnitude more, so you're limited to about 30M / 20k gas = 1500 slots
